sycl/include/sycl/aliases.hpp

```diff
@@ -106,6 +106,23 @@ using cl_half = half;
 using cl_float = float;
 using cl_double = double;
 
+namespace opencl {
+// Strictly speaking, cl_* aliases should not be defined in opencl namespace in
```

Comment:
Note for reviewers: I'm intentionally not removing `sycl::cl_*` aliases defined above, because this require more changes to both implementation and tests. I'm working on that in scope of #7915 and #1489, but I need those new aliases so I can do smooth transition.

The plan is the following:
1. add `opencl::cl_*` aliases
2. switch tests and implementation to use them (where we really need those OpenCL-specific types)
3. remove `sycl::cl_*` aliases in SYCL 2020 mode and only leave them in SYCL 1.2.1 mode